File: EndGate/tests/EndGate.Core.Tests/BoundingObjects/BoundingCircleFacts.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EndGate.Core.Assets;
using EndGate.Core.BoundingObject;
using Xunit;

namespace EndGate.Core.Tests
{
    public class BoundingCircleFacts
    {
        [Fact]
        public void AreaWorks()
        {
            var obj = new BoundingCircle(10);

            Assert.Equal(Math.Round(obj.Area()), 314);
        }

        [Fact]
        public void CircumfrenceWorks()
        {
            var obj = new BoundingCircle(10);

            Assert.Equal(Math.Round(obj.Circumfrence()), 63);
        }

        [Fact]
        public void IsCollidingWithOtherCircleWorks()
        {
            var circle1 = new BoundingCircle(10)
            {
                Position = new Vector2d(10,5)
            };

            var circle2 = new BoundingCircle(6)
            {
                Position = new Vector2d(17, 5)
            };

            Assert.True(circle1.IsCollidingWith(circle2));

            circle2.Position.X = -5;

            Assert.True(circle1.IsCollidingWith(circle2));

            circle2.Position.X = -6;

            Assert.False(circle1.IsCollidingWith(circle2));
        }

        [Fact]
        public void IsCollidingWithRectangleWorks()
        {
            var rect = new BoundingRectangle(10, 6)
            {
                Position = new Vector2d(5, 3)
            };

            var circle = new BoundingCircle(3)
            {
                Position = new Vector2d(13, 3)
            };

            Assert.False(circle.IsCollidingWith(rect));
            circle.Position.X--;
            Assert.True(circle.IsCollidingWith(rect));
            rect.Rotation = Math.PI * .5;
            rect.Position.X = 3;
            rect.Position.Y = 5;

            circle.Position.X = 8;
            circle.Position.Y = 12;

            Assert.True(circle.IsCollidingWith(rect));

        }

        [Fact]
        public void ContainsPointWorks()
        {
            var circle = new BoundingCircle(10)
            {
                Position = new Vector2d(10, 5)
            };

            Assert.False(circle.ContainsPoint(Vector2d.Zero));
            Assert.True(circle.ContainsPoint(new Vector2d(3, 3)));
        }
    }
}

File: EndGate/tests/EndGate.Core.JS.Tests/Tests/BoundingObject/BoundingCircleFacts.js
(function () {

    QUnit.module("Bounding Circle Facts");

    QUnit.test("Area Works", function () {
        var obj = new eg.Bounds.BoundingCircle(eg.Vector2d.Zero(), 10);

        QUnit.equal(Math.round(obj.Area()), 314);
    });

    QUnit.test("Circumference Works", function () {
        var obj = new eg.Bounds.BoundingCircle(eg.Vector2d.Zero(), 10);

        QUnit.equal(Math.round(obj.Circumference()), 63);
    });

    QUnit.test("Is colliding with other circle works", function () {
        var circle1 = new eg.Bounds.BoundingCircle(new eg.Vector2d(10, 5), 10);
        var circle2 = new eg.Bounds.BoundingCircle(new eg.Vector2d(17, 5), 6);

        QUnit.ok(circle1.Intersects(circle2));

        circle2.Position.X = -5;

        QUnit.ok(circle1.Intersects(circle2));

        circle2.Position.X = -6;

        QUnit.ok(!circle1.Intersects(circle2));
    });

    QUnit.test("Is colliding with other rectangle works", function () {
        var rect = new eg.Bounds.BoundingRectangle(new eg.Vector2d(5, 3), new eg.Size2d(10, 6));
        var circle = new eg.Bounds.BoundingCircle(new eg.Vector2d(14, 3), 3);

        QUnit.ok(!circle.Intersects(rect));
        circle.Position.X--;
        QUnit.ok(circle.Intersects(rect));
        rect.Rotation = Math.PI * .5;
        rect.Position.X = 3;
        rect.Position.Y = 5;

        circle.Position.X = 8;
        circle.Position.Y = 12;

        QUnit.ok(circle.Intersects(rect));

        circle = new eg.Bounds.BoundingCircle(new eg.Vector2d(156, 165), 50);
        rect = new eg.Bounds.BoundingRectangle(new eg.Vector2d(300, 200), new eg.Size2d(200, 100));

        QUnit.ok(circle.Intersects(rect));
    });

    QUnit.test("Contain point works", function () {
        var circle = new eg.Bounds.BoundingCircle(new eg.Vector2d(10, 5), 10);

        QUnit.ok(!circle.ContainsPoint(eg.Vector2d.Zero()));
        QUnit.ok(circle.ContainsPoint(new eg.Vector2d(3, 3)));
    });

})();